Validation/RecoTau/src/TauTagValidation.cc

```diff
@@ -129,7 +129,7 @@ void TauTagValidation::bookHistograms(DQMStore::IBooker& ibooker,
   ibooker.setCurrentFolder("RecoTauV/" + TauProducer_ + extensionName_ + "_Summary");
   hinfo summaryHinfo = (histoSettings_.exists("summary"))
                            ? hinfo(histoSettings_.getParameter<edm::ParameterSet>("summary"))
-                           : hinfo(21, -0.5, 20.5);
+                           : hinfo(discriminators_.size(), -0.5, discriminators_.size() - 0.5);
```

Comment:
@steggema is this change creating a non null number of bins histogram in all cases?

Reply:
For all regular cases yes, but it may be a good idea to add a check here.

Reply:
Please do so, we are having issues while merging such histograms: https://github.com/cms-sw/cmssw/issues/26727

Reply:
This was addressed by https://github.com/cms-sw/cmssw/pull/27120/commits/af33e54928329fff9a803c9389938f148dfae4fe.
Also merged with the latest master to get in sync with the meanwhile merged E/Gamma PR mentioned in the PR description).